frontend/packages/kubevirt-plugin/src/utils/url.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { k8sBasePath } from '@console/internal/module/k8s';
+
+const ELIPSIS = '…';
+
+const elipsizeLeft = (word) => `${ELIPSIS}${word}`;
+
+export const getConsoleApiBase = () => {
+  let base = k8sBasePath;
+  base = base[0] === '/' ? base.substring(1) : base; // avoid the extra slash when compose the URL by VncConsole
+  return base;
+};
+
+export const isConnectionEncrypted = () => window.location.protocol === 'https:';
+
+export const parseUrl = (url: string) => {
+  try {
+    return new URL(url);
+  } catch (e) {
+    return null;
+  }
+};
+
+const resolveOrigin = ({ hostname, origin, port }, maxHostnameParts) => {
```

Comment:
Seems like non-trivial string ops, maybe write a minimal (happy path) unit test? Same for other `resolve` functions below.

Can be done as follow-up.

Reply:
done.